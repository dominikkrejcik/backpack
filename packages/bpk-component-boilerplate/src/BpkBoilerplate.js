/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow */

import PropTypes from 'prop-types';
import React from 'react';
import { cssModules } from 'bpk-react-utils';

import STYLES from './BpkBoilerplate.scss';

const getClassName = cssModules(STYLES);

type Props = {
  title: string,
  alternate: boolean,
  className: ?string,
};
const BpkBoilerplate = (props: Props) => {
  const { alternate, className, title, ...rest } = props;
  const classNames = getClassName(
    'bpk-boilerplate',
    alternate && 'bpk-boilerplate--alternate',
    className,
  );

  return (
    <div className={classNames} {...rest}>
      {title}
    </div>
  );
};

BpkBoilerplate.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  alternate: PropTypes.bool,
};

BpkBoilerplate.defaultProps = {
  className: null,
  alternate: false,
};

export default BpkBoilerplate;
