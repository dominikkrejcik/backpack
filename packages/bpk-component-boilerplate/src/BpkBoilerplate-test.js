/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/* @flow */

import React from 'react';
import renderer from 'react-test-renderer';
import BpkBoilerplate from './BpkBoilerplate';

describe('BpkBoilerplate', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<BpkBoilerplate title="Hello world" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with the "alternate" prop', () => {
    const tree = renderer
      .create(<BpkBoilerplate title="Hello world" alternate />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support custom class names', () => {
    const tree = renderer
      .create(
        <BpkBoilerplate title="Hello world" className="custom-classname" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support arbitrary props', () => {
    const tree = renderer
      .create(<BpkBoilerplate title="Hello world" testID="123" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
