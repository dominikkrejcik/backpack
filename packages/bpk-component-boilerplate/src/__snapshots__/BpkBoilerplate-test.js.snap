// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkBoilerplate should render correctly 1`] = `
<div
  className="bpk-boilerplate"
>
  Hello world
</div>
`;

exports[`BpkBoilerplate should render correctly with the "alternate" prop 1`] = `
<div
  className="bpk-boilerplate bpk-boilerplate--alternate"
>
  Hello world
</div>
`;

exports[`BpkBoilerplate should support arbitrary props 1`] = `
<div
  className="bpk-boilerplate"
  testID="123"
>
  Hello world
</div>
`;

exports[`BpkBoilerplate should support custom class names 1`] = `
<div
  className="bpk-boilerplate custom-classname"
>
  Hello world
</div>
`;
